codec-http/src/main/java/io/netty/handler/codec/http/HttpRequestDecoder.java

```diff
@@ -320,8 +322,46 @@ private static boolean isPostMethod(final byte[] sb, int start) {
         return maybePost == POST_AS_INT;
     }
 
+    private static int firstInvalidMethodOctet(final byte[] sb, int start, int length) {
+        int end = start + length;
+        for (int i = start; i < end; i++) {
+            if (sb[i] < 0 || !VALID_HTTP_METHOD_OCTETS.get(sb[i])) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    private static final BitSet VALID_HTTP_METHOD_OCTETS = validHttpMethodOctets();
+
+    private static BitSet validHttpMethodOctets() {
+        BitSet bits = new BitSet();
+        // https://datatracker.ietf.org/doc/html/rfc9110#name-tokens
+        // token          = 1*tchar
+        // tchar          = "!" / "#" / "$" / "%" / "&" / "'" / "*"
+        //                 / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~"
+        //                 / DIGIT / ALPHA
+        //                 ; any VCHAR, except delimiters
+        //Many HTTP field values are defined using common syntax components,
+        // separated by whitespace or specific delimiting characters. Delimiters
+        // are chosen from the set of US-ASCII visual characters not allowed in
+        // a token (DQUOTE and "(),/:;<=>?@[\]{}").
+        //VCHAR          =  %x21-7E
+        bits.set(0x21, 0x7f);
+        int[] delimiters = { '(', ')', ',', '/', ':', ';', '<', '=', '>', '?', '@', '"', '[', '\\', ']', '{', '}' };
+        for (int delimiter : delimiters) {
+            bits.set(delimiter, false);
+        }
```

Comment:
I think we could just call `bits.set(...)` directly without the extra loop.

Reply:
Thank you for the suggestion.

Reply:
like this?
```
bits.set(0x21, 0x7f);
bits.clear(0x22);//DQUOTE
bits.clear(0x28, 0x2A);//()
bits.clear(0x2C);//,
bits.clear(0x3A, 0x41);//:;<=>?@
bits.clear(0x5B, 0x5E);//[\]
bits.clear(0x7B);//{
bits.clear(0x7D);//}
```

Reply:
@freeyob yes